src/Neo.Extensions/ByteArrayComparer.cs

```diff
@@ -27,28 +27,12 @@ private ByteArrayComparer(int direction)
             _direction = direction;
         }
 
-        public int Compare(byte[]? x, byte[]? y)
-        {
-            if (x == y) return 0;
-            if (x is null && y is not null)
-                return _direction > 0 ? -y.Length : y.Length;
-            if (y is null && x is not null)
-                return _direction > 0 ? x.Length : -x.Length;
-            return _direction > 0 ?
-                    CompareInternal(x!, y!) :
-                    -CompareInternal(x!, y!);
-        }
-
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static int CompareInternal(byte[] x, byte[] y)
+        public int Compare(byte[]? x, byte[]? y)
         {
-            var length = Math.Min(x.Length, y.Length);
-            for (var i = 0; i < length; i++)
-            {
-                var r = x[i].CompareTo(y[i]);
-                if (r != 0) return r;
-            }
-            return x.Length.CompareTo(y.Length);
+            if (_direction < 0)
+                return y.AsSpan().SequenceCompareTo(x.AsSpan());
```

Comment:
Add referenceEquals and null checks

Reply:
> Add referenceEquals and null checks

It will be ok even if x or/and y is null. UT has tested that.